core/src/main/java/hudson/PluginManager.java

```diff
@@ -557,7 +557,7 @@ public void run(Reactor reactor) throws Exception {
             });
 
             // All plugins are loaded. Now we can figure out who depends on who.
-            requires(PLUGINS_PREPARED).attains(COMPLETED).add("Resolving Dependant Plugins Graph", new Executable() {
+            requires(PLUGINS_STARTED, JOB_LOADED).attains(COMPLETED).add("Resolving Dependant Plugins Graph", new Executable() {
```

Comment:
Yes, this behavior would be logically correct. We are not supposed to do anything with extensions before plugins are actually loaded. I'll test the proper logic